### ISSUE TITLE
Author, license, project description, and other free text metadata

### ISSUE THREAD

**Issue by jni** [ome/ngff#213]
I am writing an image for which I wanted to include a dataset description (free text), a license, authorship, and author contributions. I expected that this would go in the "omero" transitional field, and through previous links and issues (https://github.com/ome/ngff/issues/155#issuecomment-1384163980), found this page:

https://ngff.openmicroscopy.org/latest/#omero-md

which links to this example JSON:

https://docs.openmicroscopy.org/omero/5.6.1/developers/Web/WebGateway.html#imgdata

which includes projectDescription and imageAuthor fields. Problem solved, right? Except in #155 @will-moore links to the "omero" schema:

https://github.com/ome/ngff/blob/642a7a1c6df1afcefa4c4d422920b99681d96a4a/0.4/schemas/image.schema#L53

and "meta" isn't in it at all.

Since omero is transitional, could the "meta" dict be added to the schema as optional? Having free text allows data writers to at least keep all their metadata together with their image, even if unstructured.

Certainly happy to entertain other options. 😊 Could I instead add another entry to the top-level dict (sibling to "omero") without invalidating readers?

Thank you!

**Comment by will-moore** [comment]
I don't think there's been any previous spec discussions about where to store metadata such as description, license, authorship and author contributions.
I wouldn't suggest using the `omero` block as that's really just for channels and rendering settings.
There must be plenty of existing conventions / schemas / ontologies for these kind of fields, so we don't reinvent the wheel, but I'm not familiar with this area.

There's some initial discussion about where to store custom data at https://github.com/ome/ngff/issues/68 (to use while you wait for spec changes) but that also needs some more attention.

**Comment by jni** [comment]
Thanks for the link. For free text, I'm really only looking for something that won't break existing implementations. I know how long finding a proper specification can take! 😅 Is there a chance we could define a field "custom" that is a freeform dictionary, with no compatibility guarantees? e.g. the napari "metadata" field in a layer is totally custom and has seen some good use in the community — which can then become something more formal.

**Comment by joshmoore** [comment]
Oooh, fun question. In no particular order:
 - The OME-XML does define some of these fields (Yes, yes. I hear you cringing across both the Atlantic and the Pacific) `ome-types` can make this more palatable though. :smile:
 - I'm partially tempted to say let's define a "napari" field. It would keep "omero" from being singular and provide some namespacing for what you're looking to do. More 
 - More general, though, would be a Dublin Core / Data Cite / or similar location.
 - (But really, I wanna just say "JSON-LD")

**Comment by will-moore** [comment]
In the short term I would start by picking a keyword, eg. `custom` or one of the other suggestions at https://github.com/ome/ngff/issues/68 and just start using it. Also add a comment on that issue to say what you've picked so that at least others know what you're doing and may be encouraged to pick the same - it could even be chosen to become part of the spec in the longer term. I'm not sure that "napari" is a good choice since the data you want to store isn't specific to napari (just like "omero" wasn't a good choice for rendering info as it discourages others to use it).

It would even be good to know what terms you're using within your "custom" block for author etc, since that would make a start for adding those terms to the spec too.

Much longer term "JSON-LD".

**Comment by jni** [comment]
@joshmoore I'm actually in Mallorca right now so it's just the Mediterranean between us — that's probably why you could hear it. 😂 

Anyway, I actually didn't cringe about OME-XML but at a "napari" field. 😂 OME-XML at least is a long-running standard. napari is one app and anyway it wouldn't use these fields — these are just for my own information because I feel like I should include this sort of information in the image itself.

Happy to stay out of "omero" and use "custom". Thanks @will-moore!

**Comment by lubianat** [comment]
back to this, particularly wrt to license, @clbarnes has been doing some nice work with https://github.com/clbarnes/zarr-convention-license 

see also [#NGFF > License statements in OME-Zarr files &amp; copyright liabilities @ 💬](https://imagesc.zulipchat.com/#narrow/channel/328251-NGFF/topic/License.20statements.20in.20OME-Zarr.20files.20.26.20copyright.20liabilities/near/564717316)

**Comment by lubianat** [comment]
I am thinking about this in the context of image search / findability

The [CPG gallery](https://github.com/broadinstitute/cellpainting-gallery-metadata/blob/main/harmonized_ontology.json)  has a "DOI_to_Cite" field that may be allow a sweet spot (@ErinWeisbart). 

Outsourcing some info to the DOI metadata handlers (DataCite, CrossRef etc) may make it simpler and reduce duplication of information. 

So, something like 

```
bibliography:
    name
    description
    tags
    curies
    dois_to_cite
    license_url
```

and maybe have tags and curies act as image-level baseline metadata for image search indexing or for any kind of general tagging

instantiated: 

```jsonc
  {
  "bibliography": {
        // name and description of this image / image set, not the paper title/abstract
      "name": "Attardi: Lightsheet imaging of tailbud from 18 somite stage with one frame per 2 minutes",
      "description": "In Toto Cell Tracking using Lightsheet Microscopy of the Neuromesodermal Progenitor Zone of the Zebrafish Tailbud from the 21st Somite Stage",
      // free-form indexing/search tags
      "tags": [
        "time-lapse",
        "lightsheet",
        "18-somite stage",
        "tailbud",
        "zebrafish",
        "neuromesodermal progenitor"
      ],
      // external ontology "curies", using the https://bioregistry.io/
      // preferred prefix for the resource
      "curies": ["NCBITaxon:7955", "FBbi:00000369", "ZFA:0000077"],
      // Author, institution and other information available via DOIs
      // Not duplicated in the metadata.
      "dois_to_cite": [
        // cardinality: 1..n
        { "doi": "10.1242/dev.166728", "registry": "CrossRef" },
        { "doi": "10.17867/10000117", "registry": "DataCite" }
      ],
      // A broad license URL field for the dataset. May be a standard or custom URL
      // Ideally, the detailed license information should live elsewhere (e.g. on the DataCite entry)
      // SHOULD be provided if no DataCite DOI is present
      "license_url": "https://creativecommons.org/licenses/by/4.0/"
      }
}
```

**Comment by clbarnes** [comment]
A couple of opinions to add here

1. We almost certainly don't need a new metadata standard for things like licenses, citation information, and references. We certainly don't need a new metadata standard for unstructured or free text metadata - that's what zarr attributes are for. Citation.cff seems to work well for citation information and can be serialised as JSON. For my license convention I heavily recommended SPDX but allowed some escape hatches for more baroque licenses. For references, there's CSL-JSON, hayagriva, ...
2. None of these are unique to bioimaging; if possible, we should define conventions at the zarr level and, on the NGFF side, just say "if you want to add information X your hierarchy, the `ome` object should be colocated with convention Y".
3. Relying on specific third-party services is not as good as vendoring the metadata, IMO

**Comment by lubianat** [comment]
I agreed better at Zarr level. 

On 3, a thing to note is that DOI vendors like CrossRef make checks and fixes to this kind of metadata that we certainly don't want to have Zarr tools doing, so there is a quality argument there too

**Comment by jni** [comment]
TIL about zarr conventions! Very nifty.

**Comment by jni** [comment]
> @clbarnes has been doing some nice work with [clbarnes/zarr-convention-license](https://github.com/clbarnes/zarr-convention-license?rgh-link-date=2026-01-06T09%3A53%3A29.000Z)

I agree that it's very nice 😊, but for me to consider this issue closed, I would want the convention to live within an org. Are we aiming for zarr-developers, ome, or somewhere else?
